src/server/subscriptions.ts

```diff
@@ -0,0 +1,49 @@
+import * as uuidv4 from "uuid/v4";
+import { EventEmitter } from "events";
+import { augurEmitter } from "../events";
+import { ErrorCallback } from "../types";
+
+export class Subscriptions extends EventEmitter {
+  public subscribe(eventName: string, params: any, publish: (data: {}) => void): string {
+    switch (eventName) {
+      case "MarketCreated":
+        return this.subscribeToEvent(eventName, params, publish);
+      default:
+        throw new Error(`Event ${eventName} not available for subscription`);
+    }
+  }
+
+  public unsubscribe(subscription: string): void  {
+    this.emit(`unsubscribe:${subscription}`);
+  }
+
+  public removeAllListeners(eventName?: string | symbol | undefined): this {
+    this.emit("removeAllListeners");
+    return super.removeAllListeners(eventName);
+  }
+
+  private subscribeToEvent(eventName: string, params: any, publish: (data: {}) => void): string {
+    const subscription: string = uuidv4();
+
+    const handler = (data: {}): void => {
+      this.emit(eventName, data);
+    };
+
+    augurEmitter.on(eventName, handler);
+
+    this.on(eventName, publish);
+
+    // Unsubscribe from one subscription
+    this.once(`unsubscribe:${publish}`, (): void => {
+      this.removeListener(eventName, publish);
+      augurEmitter.removeListener(eventName, handler);
+    });
```

Comment:
Same comment as w/ `handler`, I think you need to `.bind(this)` to the callback or `this.removeListener` will not work (`this` will be undefined).

Reply:
Again, ES Fat-Arrow will preserve "this" without using bind.

Internally it will transpile to something like

```
var self = this;
this.once(`unsubscribe:${eventName}`, function() {
  self.removeListener(eventName, publish);
  ...
});
```